network/gossip/libp2p/network.go

```
@@ -450,3 +476,47 @@ func (n *Network) queueSubmitFunc(message interface{}) {
 			Msg("failed to process message")
 	}
 }
+
+func (n *Network) EpochSetupPhaseStarted(newEpoch uint64, first *flow.Header) {
+	log := n.logger.With().Uint64("epoch", newEpoch).Logger()
+
+	// get the new set of IDs
+	newIDs, err := idsFromState(n.state)
+	if err != nil {
+		log.Err(err).Msg("failed to update network ids on epoch transition")
+		return
+	}
+
+	// set the network IDs to a combination of current epoch ids and next epoch ids
+	err = n.SetIDs(newIDs)
+	if err != nil {
+		log.Err(err).Msg("failed to update network ids on epoch transition")
+	}
+}
+
+// idsFromState returns the identities that the network should be using based on the current epoch phase
+func idsFromState(state protocol.ReadOnlyState) (flow.IdentityList, error) {
+
+	// epoch ids from this epoch
+	ids, err := state.Final().Identities(filter.Any)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get current epoch ids: %w", err)
+	}
+
+	// current epoch phase
+	phase, err := state.Final().Phase()
+	if err != nil {
+		log.Err(err).Msg("failed to retrieve epoch phase")
```

shouldn't we return an error at this phase and do not continue?